Rewrite integration for Spring Boot server and vscode client

BoykoAlex opened the pull request:
Integrate rewrite into STS4.

VSCode client full support.
Eclipse: no global commands, no code actions yet (investigating). Therefore, only quick fixes via problem markers are working.

@kdvolder @martinlippert Here is the PR with the proposed changes to take a look at. Will push it in tomorrow unless there are objections

martinlippert commented:
No objections to push this into main. Few thoughts here:

- the "global refactorings" should not be available or enabled by default, but internal only (for now)
- some comments in the compilation unit cache for OR seem to be old and from Eclipse JDT, should be removed
- I think all the LSP message handling code in the simple text document service should use the computeAsync way of returning the completable future, but we can take a look at this piece later to see what really needs to be put onto the async executors

BoykoAlex commented:
@martinlippert Addressed concerns from the comment above. Now global commands are shown in vscode if `boot-java.rewrite.globa-commands.on` setting is swicthed to `true`.
Will push to **main** branch tomorrow morning unless i see more comments by that time
